External/Plugins/HaXeContext/Completion/CodeComplete.cs

```diff
@@ -48,5 +49,42 @@ public override bool IsStringInterpolationStyle(ScintillaControl sci, int positi
             }
             return false;
         }
+
+        /// <inheritdoc />
+        protected override bool HandleWhiteSpaceCompletion(ScintillaControl sci, int position, string wordLeft, bool autoHide)
+        {
+            var currentClass = ASContext.Context.CurrentClass;
+            if (currentClass.Flags.HasFlag(FlagType.Abstract))
+            {
+                switch (wordLeft)
+                {
+                    case "from":
+                    case "to":
+                        return PositionIsBeforeBody(sci, position, currentClass) && HandleNewCompletion(sci, string.Empty, autoHide, wordLeft);
+                }
+            }
+            return base.HandleWhiteSpaceCompletion(sci, position, wordLeft, autoHide);
+        }
+
+        /// <summary>
+        /// Returns true if position is before body of class or member
+        /// </summary>
+        internal bool PositionIsBeforeBody(ScintillaControl sci, int position, MemberModel member)
+        {
+            var groupCount = 0;
+            var positionFrom = sci.PositionFromLine(member.LineFrom);
+            for (var i = positionFrom; i < position; i++)
+            {
+                if (sci.PositionIsOnComment(position)) continue;
+                var c = (char)sci.CharAt(i);
+                if (c == '(' || c == '<') groupCount++;
+                else if (c == ')' || (c == '>' && sci.CharAt(i - 1) != '-')) groupCount--;
+                else if (c == '{' && groupCount == 0)
+                {
+                    if (i <= position) return false;
```

Comment:
This test is always true, isn't it?

Reply:
Nope, for example:
```csharp
yield return new TestCaseData("BeforeBody_2")
                    .SetName("class Some {|}")
                    .Returns(false);
yield return new TestCaseData("BeforeBody_4")
                    .SetName("class Some {}|")
                    .Returns(false);
```

Reply:
I mean the loop is `for (var i = positionFrom; i < position; i++)` so `i` is always less than `position`

Reply:
Yes, but I do not fully understand, you see this as a mistake, or it is not immediately clear why this is done?